wp/headless-wp/includes/classes/Integrations/YoastSEO.php

```diff
@@ -162,4 +168,114 @@ public function maybe_override_sitemap_robots_url( $output ) {
 
 		return $output;
 	}
+
+	/**
+	 * Return list of query vars if the request is /yoast/v1/get_head?url and for search URL based on ?s= parameter.
+	 *
+	 * @return boolean|array False if it's not a yoast search rest api request. Search $query_vars if otherwise.
+	 */
+	public function get_yoast_search_query_vars() {
+
+		if ( ! ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
+			return false;
+		}
+
+		$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? sanitize_url( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
+
+		if ( false === strpos( $request_uri, '/yoast/v1/get_head' ) ) {
+			return false;
+		}
+
+		$url_param = isset( $_GET['url'] ) ? sanitize_url( wp_unslash( $_GET['url'] ) ) : '';
+
+		if ( filter_var( $url_param, FILTER_VALIDATE_URL ) !== false ) {
+			$query = wp_parse_url( $url_param, PHP_URL_QUERY );
+			parse_str( $query, $query_vars );
+
+			// Return query vars if search request.
+			if ( isset( $query_vars['s'] ) ) {
+				return apply_filters( 'tenup_headless_wp_search_request_query_vars', $query_vars );
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Custom helper to replace Yoast search title placeholders.
+	 * Assuming only some of the basic variables.
+	 *
+	 * @todo Support for page number.
+	 *
+	 * @param string $title The search SEO title
+	 * @param array  $query_vars The search query vars.
+	 *
+	 * @return string
+	 */
+	public function replace_yoast_search_title_placeholders( $title, $query_vars ) {
+
+		$str_replace_mapping = apply_filters(
+			'tenup_headless_wp_search_title_variables_replacments',
+			array(
+				'%%sitename%%'     => get_bloginfo( 'name' ),
+				'%%searchphrase%%' => $query_vars['s'] ?? '',
+				' %%page%%'        => '',
```

Comment:
Do you know what we need to support pages? Looks like it's not being parsed from query vars.

Reply:
not quite yet, I left a todo https://github.com/10up/headstartwp/pull/663/files#diff-39367dca2db0a8e747c2637f520848d9dd3b1321023b9eff03a75c0c51a524a7R208 comment, I can take a look next.

Reply:
@nicholasio I have pushed a change to support page number. 
However there is a limitation, in a regular WP the search title will show as "Page 2 of 10 ", we don't have the information to know how many pages there are. So it only shows "Page 2"